### ISSUE TITLE
chore: Bind `{Command,Response}Handle` to `{Client,Server}Flow`

### ISSUE THREAD

**Issue by duesee** [duesee/imap-flow#53]
Currently there is no binding and `client_flow1` can theoretically accept handles from `client_flow2`. We will have the same issue with the `Scheduler`/`TaskManager`/...

`Tag`s could be a related topic. We could make sure to never generate a `Tag` twice.

**Comment by jakoschiko** [comment]
I'm not sure why you linked `GhostCell`, it uses lifetimes and I don't think we want to use lifetimes in our case because it would cause a usability nightmare.

Would an integer from an atomic counter be good enough?

**Comment by duesee** [comment]
> Would an integer from an atomic counter be good enough?

Yeah :-)

**Comment by duesee** [comment]
When we work on this, we could also already refactor `ClientFlow.next_command_handle` (same for server) to hold a generator. This way, the type would be self-contained and we can replace ...

```rust
let handle = self.next_command_handle;
self.next_command_handle = ClientFlowCommandHandle(handle.0 + 1);
```

... with ...

```rust
let handle = self.command_handle_generator.next();
```

**Comment by jakoschiko** [comment]
Though it's nice to have different handle types for server and client for better documentation and preventing misusage during compile time.

**Comment by duesee** [comment]
Yeah, we should keep that. Only suggested to encapsulate differetly.
